x-pack/plugins/ml/public/application/components/ml_page/notifications_indicator.tsx

```diff
@@ -5,22 +5,118 @@
+import React, { FC, useEffect, useState } from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
-import { EuiFlexGroup, EuiFlexItem, EuiNotificationBadge } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiNotificationBadge,
+  EuiToolTip,
+  useEuiTheme,
+} from '@elastic/eui';
+import { timer, combineLatest, of } from 'rxjs';
+import { switchMap, filter, catchError } from 'rxjs/operators';
+import { css } from '@emotion/react';
+import { useAsObservable } from '../../hooks';
+import { NotificationsCountResponse } from '../../../../common/types/notifications';
+import { useMlKibana } from '../../contexts/kibana';
+import { useStorage } from '../../contexts/storage';
+import { ML_NOTIFICATIONS_LAST_CHECKED_AT } from '../../../../common/types/storage';
+
+const NOTIFICATIONS_CHECK_INTERVAL = 60000;
 
 export const NotificationsIndicator: FC = () => {
+  const {
+    services: {
+      mlServices: { mlApiServices },
+    },
+  } = useMlKibana();
+  const { euiTheme } = useEuiTheme();
+  const [lastCheckedAt] = useStorage(ML_NOTIFICATIONS_LAST_CHECKED_AT);
+
+  const lastCheckedAt$ = useAsObservable(lastCheckedAt);
+
+  const [notificationsCounts, setNotificationsCounts] = useState<NotificationsCountResponse>();
+
+  useEffect(function startPollingNotifications() {
+    const subscription = combineLatest([
+      lastCheckedAt$.pipe(filter((v): v is number => !!v)),
+      timer(0, NOTIFICATIONS_CHECK_INTERVAL),
+    ])
+      .pipe(
+        switchMap(([lastChecked]) =>
+          mlApiServices.notifications.countMessages$({ lastCheckedAt: lastChecked })
+        ),
+        catchError((error) => {
+          // Fail silently for now
+          return of({} as NotificationsCountResponse);
+        })
+      )
+      .subscribe((response) => {
+        setNotificationsCounts(response);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const errorsAndWarningCount =
+    (notificationsCounts?.error ?? 0) + (notificationsCounts?.warning ?? 0);
+  const hasUnread = notificationsCounts && Object.values(notificationsCounts).some((v) => v > 0);
+
   return (
-    <EuiFlexGroup alignItems={'center'} gutterSize={'xs'}>
+    <EuiFlexGroup alignItems={'center'} gutterSize={'s'}>
       <EuiFlexItem grow={false}>
         <FormattedMessage
           id="xpack.ml.navMenu.notificationsTabLinkText"
           defaultMessage="Notifications"
         />
       </EuiFlexItem>
-      {false ? (
+      {errorsAndWarningCount ? (
+        <EuiFlexItem grow={false}>
+          <EuiToolTip
+            position="right"
+            content={
+              <FormattedMessage
+                id="xpack.ml.notificationsIndicator.errorsAndWarningLabel"
+                defaultMessage="There {count, plural, one {is # notification} other {are # notification}} with the error or warning level"
```

Comment:
```suggestion
                defaultMessage="There {count, plural, one {is # notification} other {are # notification}} with error or warning level"
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/140944/commits/aebda4d040f2940b91cf432408a753674fcc063a